### PR TITLE
Implement PartialOrd for ScalarValue

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

Making it possible to compare `ScalaValue`. It is for better code reuse. Otherwise the users of `ScalaValue` might need to compare `ScalaValue` manually in their code.

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Comment by viirya** [comment]
cc @houqp @Dandandan

**Review by Dandandan — APPROVED** [review]
Thanks @viirya

File: datafusion/src/scalar.rs [hunk 1]
```diff
@@ -156,6 +157,58 @@ impl PartialEq for ScalarValue {
...
 }
 
+// manual implementation of `PartialOrd` that uses OrderedFloat to
+// get defined behavior for floating point
+impl PartialOrd for ScalarValue {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        use ScalarValue::*;
+        // This purposely doesn't have a catch-all "(_, _)" so that
+        // any newly added enum variant will require editing this list
+        // or else face a compile error
```
Comment (houqp) [inline datafusion/src/scalar.rs:167]: this comment doesn't seem to hold true anymore in this function, would be better to keep the same structure in `PartialEq` so we can also avoid catch all match here.

**Review by houqp — CHANGES_REQUESTED** [review]
Thank you @viirya !

File: datafusion/src/scalar.rs (hunk 1 above)
Comment (viirya) [reply datafusion/src/scalar.rs:167]: Good suggestion! Updated.

**Review by houqp — APPROVED** [review]
LGTM

**Comment by viirya** [comment]
Thanks @Dandandan @houqp !

**Pull request merged**